### ISSUE TITLE
Header button(s) onPress Event not firing when pressed

### ISSUE THREAD

**Issue by Mattbatchelor14** [software-mansion/react-native-screens#2551]
### Description

I am upgrading from /stack to /native-stack and when using the default back button in the stack or the headerRight method, the onPress event does not actually fire.

### Steps to reproduce

Running:
    "@react-navigation/bottom-tabs": "^7.1.3",
    "@react-navigation/elements": "^1.3.31",
    "@react-navigation/material-top-tabs": "^6.6.14",
    "@react-navigation/native": "^7.0.13",
    "@react-navigation/native-stack": "^7.1.14",
    "@react-navigation/stack": "^7.0.18",
    "expo": "^52.0.10",
    "react": "^18.3.1",
    "react-dom": "^18.3.1",
    "react-native": "0.76.2",

App.js:

import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';

const Tab = createBottomTabNavigator();
const HomeStack = createNativeStackNavigator();

  const HomeStackScreen = () => (
      <HomeStack.Navigator>
        <HomeStack.Screen name="HomeStack" component={HomeScreen} options={({title: publicSettings.company_name, 
              headerStyle: {backgroundColor: '#fff'}, headerTintColor: publicSettings.primary_hex_colour})}/>      
        <HomeStack.Screen name="Settings" component={Settings} options={({title: "Settings", headerStyle: 
             {backgroundColor: '#fff'}, headerTintColor: publicSettings.primary_hex_colour})} />
      </HomeStack.Navigator>
  )

return (
   <NavigationContainer linking={linking}>
<Tab.Navigator initialRouteName="Home" 
                      screenOptions={({ route }) => ({
                        tabBarIcon: ({ focused, color, size }) => {
                          let iconName;
                          if (route.name === 'Home') {
                            iconName = focused
                              ? 'home' : 'home';
                          } else if (route.name === 'Settings') {
                            iconName = focused ? 'settings-outline' : 'settings-outline';
                          } else if (route.name === 'RepairsScreen') {
                            iconName = focused ? 'hammer-outline' : 'hammer-outline';
                          } else if (route.name === 'RequestsScreen') {
                            iconName = focused ? 'briefcase-outline' : 'briefcase-outline';
                          } else if (route.name === 'Transactions') {
                            iconName = focused ? 'card-outline' : 'card-outline';
                          } else if (route.name === 'Messages') {
                          iconName = focused ? 'chatbox-ellipses-outline' : 'chatbox-ellipses-outline';
                        } else if (route.name === 'DocumentStack') {
                          iconName = focused ? 'documents-outline' : 'documents-outline';
                        }
                          return <Ionicons name={iconName} size={size} color={color} />;
                        },
                        tabBarActiveTintColor: publicSettings.primary_hex_colour,
                        tabBarInactiveTintColor: 'grey'
                      })}
                      >
                      <Tab.Screen name="Home" component={HomeStackScreen} options={{ headerShown: false, gestureEnabled: true } } />
                      {publicSettings.repairs_enabled ? (<Tab.Screen name="RepairsScreen" component={RepairsScreen} options={{ title: "Repairs", headerShown: false, gestureEnabled: true  }} />):(<></>)}
                      {publicSettings.requests_enabled ? (<Tab.Screen name="RequestsScreen" component={RequestsScreen} options={{ title: "Requests", headerShown: false, gestureEnabled: true }} />):(<></>)}
                      {publicSettings.transactions_enabled ? (<Tab.Screen name="Transactions" component={TransactionsScreen} options={{ title: "Transactions", headerShown: false, gestureEnabled: true }} />):(<></>)}
                      {publicSettings.document_management_enabled ? (<Tab.Screen name="DocumentStack" component={DocumentsScreen} options={({title: "Documents",headerShown: false, headerStyle: {backgroundColor: '#fff'}, headerTintColor: publicSettings.primary_hex_colour})} />):(<></>)}
                      <Tab.Screen name="Messages" component={Messages} initialParams={{ itemId: 0 }} options={({title: "Messages", tabBarBadge: messageCount, headerShown: true, headerStyle: {backgroundColor: '#fff'}, headerTintColor: publicSettings.primary_hex_colour})} />
                    </Tab.Navigator>
            </NavigationContainer>  
)
}

Inside Home.js

Using 

  useEffect(() => {
    navigation.setOptions({
      headerRight: () => (
        <Ionicons name={'cog-sharp'} size={30} color={publicSettings.primary_hex_colour} style={{marginRight: 10}} onPress={() => {navigation.navigate('Settings')}}/>
      )
    });
  }, [navigation]);

When I switch back to /native rather than /native-screens it works fine.

### Snack or a link to a repository

Unable

### Screens version

0

### React Native version

0.76.2

### Platforms

Android

### JavaScript runtime

JSC

### Workflow

Expo bare workflow

### Architecture

None

### Build type

Debug mode

### Device

Real device

### Device model

Any Android device

### Acknowledgements

Yes

**Comment by kkafar** [comment]
Hey @Mattbatchelor14, I'll close this issue in favour of: https://github.com/software-mansion/react-native-screens/issues/2219. Please track the progress there.

**Comment by Fl0p** [comment]
This reproduced for us on iOS with the New Architecture. Since the issue was closed for lack of a repro, here is the exact mechanism we measured — and the good news that **4.26.2 fixes it**.

### Symptom

A custom `headerLeft` / `headerRight` button highlights on touch (so `onPressIn` runs) but `onPress` never fires. Only extremely short taps work. In one instrumented session, presses lasting **15 ms** and **38 ms** fired, while every press of **130–150 ms** did not. That is why it looks flaky on a simulator (a mouse click can be short enough) and effectively always broken on a device with a finger.

### Root cause

`Pressability` captures the press region on responder grant via `measure()`, which on Fabric reports the position from the shadow tree. For a header subview that position does not match where UIKit actually placed the view.

Instrumented on a physical device — iOS 16.7.16, RN 0.86.0, `react-native-screens` 4.25.2, `@react-navigation/native-stack` 7.17.7 — for a 44×44 button whose real origin was `(16, 44)`:

```
measure()   ->  pageXY: [16, -148],  size: [44, 44]
touch       ->  pageXY: [35.7, 53.7]
                insideX: true,  insideY: false
```

Stable across taps: `pageY` was off by exactly **192 pt**. The consequence is that the first `onResponderMove` after grant concludes the finger is outside the button, Pressability transitions to `RESPONDER_ACTIVE_PRESS_OUT`, and the release then emits `onPressOut` **without** `onPress`. A tap short enough to emit no `move` at all survives — hence the timing dependency.

Worth noting for anyone debugging this: the gesture is *not* being stolen. `onTouchStart` and `onTouchEnd` both arrive normally and no `onTouchCancel` is ever emitted. Only the region check is wrong.

### Fixed in 4.26.2

The same build, same device, upgraded from 4.25.2 to 4.26.2 with no other change:

```
measure()   ->  pageXY: [16, 44],  size: [44, 44]
touch       ->  pageXY: [52, 55.3]
                insideX: true,  insideY: true
```

Every tap now fires `onPress` on the first try, on both the left and the right header slots.

The likely reason is that `-[RNSScreenStackHeaderSubview initWithFrame:]` now sets `_synchronousShadowStateUpdatesEnabled = YES`, so the frame that `updateShadowStateWithFrame:` reports back lands in the shadow tree immediately instead of asynchronously — and the shadow tree is exactly what `measure()` reads. (I have not bisected that, so treat the attribution as a hypothesis; the before/after behaviour itself is measured.)

**tl;dr for anyone landing here from a search:** if custom header buttons highlight but do not fire `onPress` on the New Architecture, upgrade to `react-native-screens` 4.26.2.
